--- autoscan-network-1.40.old/src/autoscan-network/Modules.c	2008-12-29 19:30:27.000000000 +0100
+++ autoscan-network-1.40/src/autoscan-network/Modules.c	2008-12-29 19:29:40.000000000 +0100
@@ -266,7 +266,7 @@
 #ifdef         __NESSUS_ENABLE
 	ERROR_if (Equipement_Selectionner == NULL);
 	Check_Equipement_Up(Equipement_Selectionner,Reseau_Selectionner);
-	if (Modules_Actuel != MODULES_NESSUS_MESSAGE) {
+	if (Equipement_Selectionner->Modules_Actuel != MODULES_NESSUS_MESSAGE) {
 		gtk_widget_set_sensitive (combobox_liste_Modules, FALSE);
 		gtk_widget_set_sensitive (button_Refresh, FALSE);
 		Destruction_Modules (FALSE);
diff -ur autoscan-network-1.40.old/src/autoscan-network/Interface/Nessus_Message_Callbacks.c autoscan-network-1.40/src/autoscan-network/Interface/Nessus_Message_Callbacks.c
--- autoscan-network-1.40.old/src/autoscan-network/Interface/Nessus_Message_Callbacks.c	2008-12-29 19:33:19.000000000 +0100
+++ autoscan-network-1.40/src/autoscan-network/Interface/Nessus_Message_Callbacks.c	2008-12-29 19:42:18.000000000 +0100
@@ -39,12 +39,15 @@
 #include "Root_Interface.h"
 #include "../Modules.h"
 #include "../Debug.h"
+#include "../Equipement.h"
+
 
 void on_button_Nessus_clicked (GtkButton * button, gpointer user_data)
 {
 	DEBUG_ACTION_USER();
+    ERROR_if (Equipement_Selectionner == NULL);
 	gtk_combo_box_set_active (GTK_COMBO_BOX (combobox_liste_Modules), 0);
-	if (Modules_Actuel != MODULES_LISTE_PORTS)
+	if (Equipement_Selectionner->Modules_Actuel != MODULES_LISTE_PORTS)
 		Initialisation_Liste_ports();
 }
 
