--- old/src/linux-2.6.29/drivers/staging/rt2870/rt2870.h	2009-03-24 00:12:14.000000000 +0100
+++ new/src/linux-2.6.29/drivers/staging/rt2870/rt2870.h	2009-03-27 21:18:12.000000000 +0100
@@ -104,6 +104,7 @@
 	{USB_DEVICE(0x14B2,0x3C07)}, /* AL */			\
 	{USB_DEVICE(0x14B2,0x3C12)}, /* AL */           \
 	{USB_DEVICE(0x050D,0x8053)}, /* Belkin */		\
+	{USB_DEVICE(0x050D,0x815C)}, /* Belkin */		\
 	{USB_DEVICE(0x14B2,0x3C23)}, /* Airlink */		\
 	{USB_DEVICE(0x14B2,0x3C27)}, /* Airlink */		\
 	{USB_DEVICE(0x07AA,0x002F)}, /* Corega */		\
 
