Index: setup.py
===================================================================
--- setup.py	(revision 69)
+++ setup.py	(working copy)
@@ -25,7 +25,7 @@
 import sys, subprocess, re, os
 
 # Options to use for all modules
-EXTRA_COMPILE_ARGS = ['-O2', '-Werror']
+EXTRA_COMPILE_ARGS = ['-O2']
 LIBRARY_DIRS = ['/usr/lib']
 INCLUDE_DIRS = ['/usr/include','/usr/local/include']
 
Index: _cpyrit/_cpyrit_cpu.c
===================================================================
--- _cpyrit/_cpyrit_cpu.c	(revision 69)
+++ _cpyrit/_cpyrit_cpu.c	(working copy)
@@ -18,7 +18,7 @@
 #    along with Pyrit.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <python2.5/Python.h>
+#include <python2.6/Python.h>
 #include <openssl/hmac.h>
 #include <openssl/sha.h>
 #include <sys/ucontext.h>
