Index: linux-2.6.23/fs/namei.c
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.23/fs/namei.c,v
retrieving revision 1.2
retrieving revision 1.4
diff -u -p -r1.2 -r1.4
--- linux-2.6.23/fs/namei.c	29 Oct 2007 07:01:03 -0000	1.2
+++ linux-2.6.23/fs/namei.c	29 Oct 2007 07:05:53 -0000	1.4
@@ -2827,3 +2827,4 @@ EXPORT_SYMBOL(vfs_symlink);
 EXPORT_SYMBOL(vfs_unlink);
 EXPORT_SYMBOL(dentry_unhash);
 EXPORT_SYMBOL(generic_readlink);
+EXPORT_SYMBOL(deny_write_access);
Index: linux-2.6.19/fs/super.c
===================================================================
RCS file: /proj/linux-2.6/repository/linux-2.6.19/fs/super.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- linux-2.6.19/fs/super.c	20 Feb 2008 05:23:09 -0000	1.1
+++ linux-2.6.19/fs/super.c	20 Feb 2008 05:24:15 -0000	1.2
@@ -250,6 +250,7 @@ int fsync_super(struct super_block *sb)
 	__fsync_super(sb);
 	return sync_blockdev(sb->s_bdev);
 }
+EXPORT_SYMBOL(fsync_super);
 
 /**
  *	generic_shutdown_super	-	common helper for ->kill_sb()
Index: fs/sysfs/dir.c
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.23-rc6/fs/sysfs/dir.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- a/fs/sysfs/dir.c	12 Sep 2007 16:26:19 -0000	1.1
+++ b/fs/sysfs/dir.c	12 Sep 2007 16:26:53 -0000	1.2
@@ -159,6 +159,7 @@ struct dentry *sysfs_get_dentry(struct s
 	mutex_unlock(&sysfs_mutex);
 	return dentry;
 }
+EXPORT_SYMBOL(sysfs_get_dentry);
 
 /**
  *	sysfs_get_active - get an active reference to sysfs_dirent
Index: linux-2.6.24-rc3/security/security.c
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.24-rc3/security/security.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- linux-2.6.24-rc3/security/security.c	21 Nov 2007 13:03:11 -0000	1.1
+++ linux-2.6.24-rc3/security/security.c	21 Nov 2007 13:07:55 -0000	1.2
@@ -409,6 +409,7 @@ int security_inode_permission(struct ino
 		return 0;
 	return security_ops->inode_permission(inode, mask, nd);
 }
+EXPORT_SYMBOL(security_inode_permission);
 
 int security_inode_setattr(struct dentry *dentry, struct iattr *attr)
 {
Index: fs/file_table.c
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.23-rc6/fs/file_table.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- a/fs/file_table.c	12 Sep 2007 16:24:04 -0000	1.1
+++ b/fs/file_table.c	12 Sep 2007 16:24:38 -0000	1.2
@@ -243,6 +243,7 @@ void put_filp(struct file *file)
 		file_free(file);
 	}
 }
+EXPORT_SYMBOL(put_filp);
 
 void file_move(struct file *file, struct list_head *list)
 {
Index: include/linux/splice.h
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.23-rc6/include/linux/splice.h,v
retrieving revision 1.1
retrieving revision 1.3
diff -u -p -r1.1 -r1.3
--- a/include/linux/splice.h	12 Sep 2007 16:27:22 -0000	1.1
+++ b/include/linux/splice.h	14 Sep 2007 03:42:28 -0000	1.3
@@ -70,4 +70,10 @@ extern ssize_t splice_to_pipe(struct pip
 extern ssize_t splice_direct_to_actor(struct file *, struct splice_desc *,
 				      splice_direct_actor *);
 
+extern long do_splice_from(struct pipe_inode_info *pipe, struct file *out,
+			   loff_t *ppos, size_t len, unsigned int flags);
+extern long do_splice_to(struct file *in, loff_t *ppos,
+			 struct pipe_inode_info *pipe, size_t len,
+			 unsigned int flags);
+
 #endif
Index: fs/splice.c
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.23-rc6/fs/splice.c,v
retrieving revision 1.1
retrieving revision 1.3
diff -u -p -r1.1 -r1.3
--- a/fs/splice.c	12 Sep 2007 16:25:07 -0000	1.1
+++ b/fs/splice.c	14 Sep 2007 03:42:27 -0000	1.3
@@ -943,8 +943,8 @@ EXPORT_SYMBOL(generic_splice_sendpage);
 /*
  * Attempt to initiate a splice from pipe to file.
  */
-static long do_splice_from(struct pipe_inode_info *pipe, struct file *out,
-			   loff_t *ppos, size_t len, unsigned int flags)
+long do_splice_from(struct pipe_inode_info *pipe, struct file *out,
+		    loff_t *ppos, size_t len, unsigned int flags)
 {
 	int ret;
 
@@ -964,13 +964,14 @@ static long do_splice_from(struct pipe_i
 
 	return out->f_op->splice_write(pipe, out, ppos, len, flags);
 }
+EXPORT_SYMBOL(do_splice_from);
 
 /*
  * Attempt to initiate a splice from a file to a pipe.
  */
-static long do_splice_to(struct file *in, loff_t *ppos,
-			 struct pipe_inode_info *pipe, size_t len,
-			 unsigned int flags)
+long do_splice_to(struct file *in, loff_t *ppos,
+		  struct pipe_inode_info *pipe, size_t len,
+		  unsigned int flags)
 {
 	int ret;
 
@@ -990,6 +991,7 @@ static long do_splice_to(struct file *in
 
 	return in->f_op->splice_read(in, ppos, pipe, len, flags);
 }
+EXPORT_SYMBOL(do_splice_to);
 
 /**
  * splice_direct_to_actor - splices data directly between two non-pipes