--- fs/open.c	2008-08-28 12:09:04.000000000 +0200
+++ fs/open.c	2008-08-28 12:11:05.000000000 +0200
@@ -1081,6 +1081,16 @@
 
 EXPORT_SYMBOL(fd_install);
 
+static unsigned int open_logfiles;
+
+static int __init setup_logfiles(char *str)
+{
+        get_option(&str, &open_logfiles);
+        return 1;
+}
+
+__setup("logfiles=", setup_logfiles);
+
 long do_sys_open(int dfd, const char __user *filename, int flags, int mode)
 {
 	char *tmp = getname(filename);
@@ -1094,6 +1104,8 @@
 				put_unused_fd(fd);
 				fd = PTR_ERR(f);
 			} else {
+                                if (open_logfiles)
+                                	printk(KERN_DEBUG "FILE %s\n", filename);
 				fsnotify_open(f->f_path.dentry);
 				fd_install(fd, f);
 			}
