--- fs/squashfs/inode.c.__orig__	2008-12-28 15:34:48.000000000 +0100
+++ fs/squashfs/inode.c	2008-12-28 15:38:31.000000000 +0100
@@ -675,31 +675,14 @@
 	unsigned int inode_number)
 {
 	squashfs_inode_t inode;
-	struct inode *i;
-	struct dentry *dentry;
 
 	TRACE("Entered squashfs_export_iget\n");
 
 	inode = squashfs_inode_lookup(s, inode_number);
-	if(inode == SQUASHFS_INVALID_BLK) {
-		dentry = ERR_PTR(-ENOENT);
-		goto failure;
-	}
-
-	i = squashfs_iget(s, inode, inode_number);
-	if(i == NULL) {
-		dentry = ERR_PTR(-EACCES);
-		goto failure;
-	}
-
-	dentry = d_alloc_anon(i);
-	if (dentry == NULL) {
-		iput(i);
-		dentry = ERR_PTR(-ENOMEM);
-	}
-
-failure:
-	return dentry;
+	if(inode == SQUASHFS_INVALID_BLK)
+		return ERR_PTR(-ENOENT);
+	
+	return d_obtain_alias(squashfs_iget(s, inode, inode_number));
 }
 
 
