--- unicornscan-0.4.7.old/src/unilib/socktrans.c	2006-10-18 18:57:05.000000000 +0200
+++ unicornscan-0.4.7/src/unilib/socktrans.c	2008-12-31 12:01:02.000000000 +0100
@@ -32,6 +32,14 @@
 #include <unilib/output.h>
 #include <unilib/xmalloc.h>
 
+/* Dirty hack to avoid errors (from bits/socket.h) */
+struct ucred
+{
+  pid_t pid;                    /* PID of sending process.  */
+  uid_t uid;                    /* UID of sending process.  */
+  gid_t gid;                    /* GID of sending process.  */
+};
+
 static uint16_t lbind=BINDPORT_START;
 
 static void accept_timeout(int ); /* signal handler */
